### PR TITLE
[Stack 4/4] Shared parity layer for CRM and website

### LINKED ISSUE #7: Skeleton
### Description
Add a skeleton loader component to match the Figma design. This will be used as a placeholder while the actual content is loading in the UI.

Tasks

- [ ]  Create a Skeleton component in the components folder.
- [ ]  Implement skeleton variants (e.g., card, text, avatar) based on Figma design.
- [ ]  Integrate the skeleton component into existing components where loading states are needed.
- [ ]  Add unit tests to verify the skeleton renders correctly.
- [ ]  Add Storybook stories for the skeleton component.

Acceptance Criteria

- [ ]  Skeleton component exists in the components folder.
- [ ]  All skeleton variants match the Figma design.
- [ ]  Components with loading states show the skeleton appropriately.
- [ ]  Unit tests pass for skeleton rendering.
- [ ]  Storybook stories are available and demonstrate all skeleton variants.

Labels: frontend, UI, enhancement, skeleton

Remember to use helpful labels and milestones

### LINKED ISSUE #51: Full test coverage across the test pyramid (unit → e2e/visual/perf/memleak)
## Goal

Bring `@vilnacrm/ui-toolkit` to full, enforced test coverage across the whole test pyramid, mirroring the CRM repo's per-type conventions. Work is staged on branch `review-4-test-coverage` (stacked on PR #46's parity layer) and will land as a separate PR — split across two days, ~50% each.

## Scope (test pyramid)

- [ ] **Unit** — 100% line/branch coverage on all components; 100% gate enforced
- [ ] **Integration** — composition-component tests; 100% gate, scoped to composition components
- [ ] **Lint** — adopt CRM ESLint config; testing-library compliance
- [ ] **Mutation** — Stryker score ≥ 98% (currently raised 73% → 98%)
- [ ] **E2E** — behavioral a11y flows (Playwright)
- [ ] **Visual** — Chromium visual regression; 100% story coverage
- [ ] **Perf / Lighthouse** — Lighthouse CI budgets vs Storybook
- [ ] **Memory leak** — memlab scenarios vs Storybook
- [ ] **Stories** — 100% story coverage enforced across e2e + visual suites

## Already staged on `review-4-test-coverage` (12 commits, +5,700 / −603 over 141 files)

| Phase | Commit |
|---|---|
| Foundation scaffolding | `561a66e` |
| Unit 100% + gate | `d09f9bf` |
| Integration composition | `b5847b6` |
| Lint / testing-library | `2f55d1a` |
| Mutation 73%→98% | `82a25ca` |
| a11y reduced-motion + stories | `d735ae8` |
| E2E + visual (Phase 5) | `94c91a0` |
| Checkbox fix + state-grid visual | `f5f1c07` |
| Typography medium15 fix | `a630d3a` |
| Perf + memory (Phase 6) | `b70af1e` |
| Integration coverage gate | `4747c71` |
| Story coverage (e2e + visual) | `eeec9b9` |

## Notes

- Depends on PR #46 (parity layer) — the tests exercise the parity components. Rebase onto `main` once #46 merges, or land as a stacked PR with base `review-4-shared-parity`.
- Three commits are `feat`/`fix` rather than pure `test` (`d735ae8`, `f5f1c07`, `a630d3a`); confirm at review whether any belong in #46 instead.
- Known residuals from the #46 merge to revisit here: ~69 eslint errors (stricter `@typescript-eslint/typedef` / `explicit-function-return-type`) and a half-migrated `UiFooter` (`index.ts` shadows the parity `index.tsx`).

## Delivery plan

- Day 1: ~50% of the commits
- Day 2: remaining ~50%, open the PR

### PR DESCRIPTION
**Pull request by RudoiDmytro**
## Summary
This stacked follow-up carries the shared-layer parity work that sits above PR #39.

It contains:
- CRM-side shared components (UiContainer, UiForm, UiBackToMain, skeleton primitives, AuthSkeleton)
- widened shared component contracts for button, text-field, typography, checkbox, link, and tooltip surfaces
- explicit CRM and website token exports for breakpoints and color themes
- neutral shared-layer updates for the exported card/footer surface already introduced earlier in the stack
- unit coverage for the new shared parity surface

## Stack
- Previous PR: #39
- This PR base: review-3-contract-baseline

## Notes
- Verification on this stacked branch is currently limited. A direct bun x tsc --noEmit run against this older stack snapshot still fails broadly because the branch dependency and test configuration diverges from the current parity worktree, so the strongest verification remains the clean gate on the current worktree where this split originated.
- PR #39 was updated separately to expose the composed layer and add Layout on its own branch.

### CODE REVIEW

**Comment by RudoiDmytro** [comment]
@coderabbitai review

**Comment by RudoiDmytro** [comment]
## Review threads addressed — `4352d24` (+ `26d0776`, `eb829b9`)

**Fixed in code:**
- **UiTooltip** `triggerLabel` now forwarded to the wrapper (was dropped at the public boundary) — restores the accessible name for icon triggers (WCAG 4.1.2). Added a regression test.
- **build.config** externalizes only peer deps instead of `packages: 'external'`, so Swiper's carousel CSS is bundled back into `build/index.css` (`styles.css` export). Verified `index.css` 1.2K → 12.7K.
- **package.json** dropped the `peerDependenciesMeta` optional flags — react-hook-form / i18next / react-i18next are statically imported by the single-entry bundle, so they're required.
- **UiLink** always merges `noopener noreferrer` for `target="_blank"` even with a custom `rel` (anti tab-nabbing), token-deduped.
- **UiBackToMain** `aria-label` only set when `label` is a string (avoids `"[object Object]"`).
- **UiInput** `onInput` typed as native `TextFieldProps['onInput']`; unsafe cast removed.
- **UiTextFieldForm** composes consumer `onChange`/`onBlur` with RHF's instead of overwriting them.
- **UiCardItem** `SMALL_CARD_TEXT` typed as `CardType`.
- **UiColorTheme.test** asserts `strokeDanger` parity across website/CRM presets.
- **CardSwiper** redundant `className="swiper-wrapper"` removed.

**No change (with rationale, replied inline & resolved):**
- `newTabLabel` English default kept — removing it regresses the new-tab a11y hint; consumers localize via the prop.
- `fonts.css` font-family quoting — no stylelint is configured in this repo, so there's no failing gate.

**Deferred to the memory-leak test batch** (separate, ≤150-file daily push): the 5 memlab scenario comments (`runMemlabTests` path bug, selector waits, scenario determinism) land with the Phase-6 memory-leak work and its Chrome-install fix.

All changes: tsc clean, eslint 0 errors, prettier clean, 36 unit suites / 120 tests pass. Accessibility-lead reviewed every UI change.

**Comment by RudoiDmytro** [comment]
@coderabbitai review

**Comment by RudoiDmytro** [comment]
@coderabbitai review

File: tests/memory-leak/tests/checkboxToggle.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    Promise.resolve()
+  );
+}
+
+async function back(page) {
+  const isChecked = await page.$eval(checkboxSelector, element => element.checked);
+  if (isChecked) {
+    await page.click(checkboxSelector);
+  }
+}
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/memory-leak/tests/checkboxToggle.js:42]: Done in 9f8f054 — `back` now waits via `page.waitForFunction` until the checkbox is unchecked before returning.

File: tests/e2e/stories.smoke.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const response = await request.get(`${baseURL}/index.json`);
+  expect(response.ok()).toBeTruthy();
+
+  const index: { entries: Record<string, IndexEntry> } = await response.json();
+  const liveStoryIds: string[] = Object.values(index.entries)
+    .filter(entry => entry.type === 'story')
+    .map(entry => entry.id)
+    .sort();
+  const manifestIds: string[] = (stories as StoryEntry[]).map(entry => entry.id).sort();
+
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/e2e/stories.smoke.spec.ts:46]: Done in 9f8f054 — both `liveStoryIds` and `manifestIds` now sort with `(a, b) => a.localeCompare(b)`, clearing the Sonar violations.

File: tests/e2e/layout.spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  test('exposes banner, main and contentinfo landmarks with a single h1', async ({ page }) => {
+    await gotoStory(page, 'uicomponents-layout--default');
+
+    await expect(page.getByRole('banner')).toBeVisible();
+    await expect(page.getByRole('main')).toBeVisible();
+    await expect(page.getByRole('contentinfo')).toBeVisible();
+    await expect(page.getByRole('heading', { level: 1, name: 'Vilna Toolkit' })).toBeVisible();
+    await expect(page.getByRole('heading', { level: 2, name: 'Dashboard' })).toBeVisible();
+    await expect(page.getByRole('button', { name: 'Continue' })).toBeVisible();
+  });
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/e2e/layout.spec.ts:16]: Done in 9f8f054 — added `await expect(page.getByRole('heading', { level: 1 })).toHaveCount(1)` so duplicate h1 regressions are caught.

File: tests/memory-leak/tests/formFill.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+async function action(page) {
+  await page.waitForSelector(emailInputSelector, { visible: true });
+  await page.click(emailInputSelector);
+  await page.type(emailInputSelector, 'leak-check@example.com');
+}
+
+async function back(page) {
+  await page.click(emailInputSelector, { clickCount: 3 });
+  await page.keyboard.press('Backspace');
+}
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/memory-leak/tests/formFill.js:27]: Done in 9f8f054 — `back` now waits via `page.waitForFunction` until the email field value is empty before returning.

File: tests/integration/components/UiCardList.integration.test.tsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    const reactNodeCard: UiCardItemData = {
+      ...SMALL_CARDLIST_ARRAY[0],
+      text: <span>Custom node body</span>,
+    };
+
+    // A smallCard first item makes CardGrid pick the small grid (vs the large
+    // grid); a ReactNode `text` exercises CardContent's non-string render branch.
+    render(<UiCardList cardList={[reactNodeCard]} />);
+
+    expect(screen.getByText('Custom node body')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/integration/components/UiCardList.integration.test.tsx:227]: Done in 9f8f054 — added `expect(screen.queryByTestId('swiper')).not.toBeInTheDocument()` to pin this edge case to the grid path.

File: tests/memory-leak/tests/tooltipToggle.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const ScenarioBuilder = require('../utils/ScenarioBuilder');
+
+const scenarioBuilder = new ScenarioBuilder();
+
+const triggerSelector = '[role="button"]';
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/memory-leak/tests/tooltipToggle.js]: Done in 9f8f054 — replaced the generic `[role="button"]` with the trigger's stable accessible name via puppeteer's ARIA handler: `::-p-aria([name="Hello World!"][role="button"])`. Avoided a test-only `data-testid` to keep the component clean.

**Comment by RudoiDmytro** [comment]
@coderabbitai review

**Comment by RudoiDmytro** [comment]
@cubic-dev-ai review

**Comment by RudoiDmytro** [comment]
@cubic-dev-ai review

**Comment by RudoiDmytro** [comment]
@cubic-dev-ai review

**Review by Kravalg — APPROVED** [review]
